src/main/java/com/xeeva/catalog/CostCenterPage.java

```diff
@@ -152,16 +267,172 @@ public void change_CC(String changeType,String CC){
+		TestReporter.assertTrue(getReqByDateMessage.contains("date has been updated."), 
+				"Required by date has been updated at CC-Level.");
+
+	}
+
+	/**
+	 * @summary: Method to change the RequiredBy value at linelevel.
+	 * @author: Praveen Namburi, @version: Created 20-09-2016.
+	 */
+	public void change_RequiredByAtLineLevel(){
+		// after cart check out, application taking time to load cost center page 
+		lblCC.syncVisible(30, false);
+		lblCC.isDisplayed();
+		driver.setPageTimeout(3);
+
+		String getReqByDate_beforeChange = getReqByDateAtLineLevel();
+		TestReporter.log("Get ReqByDate_before Change: " + getReqByDate_beforeChange);
+
+		driver.executeJavaScript("arguments[0].click();", dateCCLineLevel.get(0));
+		List<WebElement> nextMonthArrows = driver.findElements(By.xpath("html/body/div[@class='calendar']/"
+				+ "table/thead/tr[2]/td[4]"));
+		int getNextMonthArrowsSize = nextMonthArrows.size();
+		System.out.println("Next month arrows size: "+getNextMonthArrowsSize);
+		for(WebElement nextMonth : nextMonthArrows){
+			driver.setPageTimeout(2);
+			nextMonth.click();
+			driver.setPageTimeout(2);
+			List<WebElement> selectDates = driver.findElements(By.xpath("html/body/div[@class='calendar']/"
+					+ "table/tbody/tr[4]/td[@class='day']"));
+			for(WebElement selWeekDate : selectDates){
+				selWeekDate.click();
+				break;
+			}
+		}
+
+		//Verify the message -'The RequiredBy date has been updated' is displayed.
+		verify_RequiredByDateUpdated_atCCLevel();
+		driver.setPageTimeout(2);
+
+		List<WebElement> inputDates = dateCCInputLineLevel;
+		int getInputDatesSize = inputDates.size();
+		TestReporter.log("Total no. of input dates available at CC_line-level: " + getInputDatesSize);
+		//Iterate through each ReqBydate element and get text,
+		//verify the expected date with actual values.
+		for(WebElement inputDate : inputDates){
+			String getModifiedDate = inputDate.getAttribute("value");
+			TestReporter.log("Get ReqByDate after change: "+getModifiedDate);
+			TestReporter.assertNotEquals(getReqByDate_beforeChange, getModifiedDate, 
+					"Date has been updated at line-level for the id: - ["+inputDate.getAttribute("id")+"]");
+			break;
+		}
+
+	}
+
+	/**
+	 * @summary: Method to change the RequiredBy value at Header level.
+	 * @author: Praveen Namburi, @version: Created 22-09-2016.
+	 * @param daysOut
+	 */
+	public void change_RequiredByAtHeaderLevel(){
+		// after cart check out, application taking time to load cost center page 
+		lblCC.syncVisible(30, false);
+		lblCC.isDisplayed();
+		driver.setPageTimeout(3);
+		//To click on calender at CC_HeaderLevel.
+		driver.executeJavaScript("arguments[0].click();", lnkDateCC_HeaderLvel);
+		List<WebElement> nextMonthArrows = driver.findElements(By.xpath("html/body/div[@class='calendar']/"
+				+ "table/thead/tr[2]/td[4]"));
+		int loopCount=0;
+		for(WebElement nextMonth : nextMonthArrows){
+			driver.setPageTimeout(2);

```

Comment:
Here you need to just remove the driver.setPageTimeout(2).  You can't really use an the setElementTimeout as it isn't applicable - only works when using a driver.findElementBy() or driver.findElementsBy() and as you are using the page factory element you've defined (nextMonth) that is already stored in memory and doesn't use the findElement method.